src/dmd/cparse.d

```diff
@@ -2623,7 +2623,7 @@ final class CParser(AST) : Parser!AST
             nextToken();
         }
 
-        auto etag = new AST.EnumDeclaration(loc, tag, AST.Type.tint32);
+        auto etag = new AST.EnumDeclaration(loc, toTagIdentifier(tag), AST.Type.tint32);
```

Comment:
FYI, this should be pooled in a symtab to avoid false duplicate errors.

https://issues.dlang.org/show_bug.cgi?id=21932

Reply:
That's the job of the semantic routines, not the parser.

Reply:
As we know you can't use a symbol without declaration or prototype, type/symbol merging in the parser is safe so long as it's not a redefinition (and saves memory). This is also required for attributes to work too.